compiler/lquantizer/include/Quantizer.h

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __QUANTIZER_H__
+#define __QUANTIZER_H__
+
+#include <luci/IR/Module.h>
+#include <luci_interpreter/Interpreter.h>
+
+#include "QuantizationObservers.h"
+
+#include <memory>
+
+namespace lquantizer
+{
+
+class Quantizer
+{
+public:
+  Quantizer() = default;
+
+  Quantizer(uint32_t bits_per_input, uint32_t bit_ber_weight)
+    : _input_encoding_bits(bits_per_input), _weights_encoding_bits(bit_ber_weight)
+  {
+    // Do nothing
+  }
+
+  ~Quantizer() = default;
+
+  void initialize(const std::string &input_model_path);
+
+  void train();
```

Comment:
This function will call 2 private functions:
1. `train_weights()`
2. `train_input()`

Reply:
TODO introduce them

Reply:
May be better to incapsulate `uptrain` in `train`  function too latter